mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraImapUidDAOTest.java

```diff
@@ -0,0 +1,123 @@
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox.cassandra.mail;
+
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.apache.james.backends.cassandra.CassandraCluster;
+import org.apache.james.mailbox.MessageUid;
+import org.apache.james.mailbox.cassandra.CassandraId;
+import org.apache.james.mailbox.cassandra.CassandraMessageId;
+import org.apache.james.mailbox.cassandra.modules.CassandraMessageModule;
+import org.apache.james.mailbox.model.ComposedMessageId;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.datastax.driver.core.utils.UUIDs;
+import com.github.steveash.guavate.Guavate;
+
+public class CassandraImapUidDAOTest {
+
+    private CassandraCluster cassandra;
+
+    private CassandraImapUidDAO testee;
+
+    @Before
+    public void setUp() {
+        cassandra = CassandraCluster.create(new CassandraMessageModule());
+        cassandra.ensureAllTables();
+
+        testee = new CassandraImapUidDAO(cassandra.getConf());
+    }
+
+    @After
+    public void tearDown() {
+        cassandra.clearAllTables();
+    }
+
+    @Test
+    public void deleteShouldNotThrowWhenRowDoesntExist() {
+        testee.delete(CassandraMessageId.of(UUIDs.timeBased()), CassandraId.timeBased())
+            .join();
+    }
+
+    @Test
+    public void deleteShouldDeleteWhenRowExists() {
+        CassandraMessageId messageId = CassandraMessageId.of(UUIDs.timeBased());
+        CassandraId mailboxId = CassandraId.timeBased();
+        MessageUid messageUid = MessageUid.of(1);
+        testee.insert(messageId, mailboxId, messageUid).join();
+        ComposedMessageId expectedComposedMessageId = new ComposedMessageId(mailboxId, messageId, messageUid);
+        Stream<ComposedMessageId> insertedMessages = testee.retrieve(messageId, Optional.of(mailboxId));
+        assertThat(insertedMessages.collect(Guavate.toImmutableList())).containsOnly(expectedComposedMessageId);
+
+        testee.delete(messageId, mailboxId).join();
+
+        Stream<ComposedMessageId> messages = testee.retrieve(messageId, Optional.of(mailboxId));
+        assertThat(messages.collect(Guavate.toImmutableList())).isEmpty();
+    }
+
+    @Test
+    public void insertShouldInsert() {
+        CassandraMessageId messageId = CassandraMessageId.of(UUIDs.timeBased());
+        CassandraId mailboxId = CassandraId.timeBased();
+        MessageUid messageUid = MessageUid.of(1);
+
+        testee.insert(messageId, mailboxId, messageUid).join();
+
+        ComposedMessageId expectedComposedMessageId = new ComposedMessageId(mailboxId, messageId, messageUid);
+        Stream<ComposedMessageId> messages = testee.retrieve(messageId, Optional.of(mailboxId));
+        assertThat(messages.collect(Guavate.toImmutableList())).containsOnly(expectedComposedMessageId);
```

Comment:
I guess he can not use assertj 3 because store relies on assertj 1, and he had conflicts